restaurant and menu_item models set up along with tests and schema

eport01 opened the pull request:
Database and relationships created. Restaurant and MenuItem models created along with basic tests. All tests pass

eport01 committed:
restaurant and menu_item models set up along with tests and schema

eport01 committed:
seed db with faker data for testing purposes

eport01 committed:
get all restaurants endpoint and spec file complete

eport01 committed:
update get all restaurants endpoint spec file

eport01 committed:
create a new restaurant endpoint and tests complete

eport01 committed:
update create restaurant spec

Yuji3000 commented on db/migrate/20230204044855_create_menu_items.rb:
```diff
@@ -0,0 +1,15 @@
+      t.belongs_to :restaurant, foreign_key: true
+      t.string :name
+      t.string :description
+      t.string :allergens
```
Allergens should be changed to "tags"

eport01 replied:
Updated!

Yuji3000 commented on db/migrate/20230204044855_create_menu_items.rb:
```diff
@@ -0,0 +1,15 @@
+      t.string :allergens
+      t.string :category
+      t.string :image
+      t.string :price
```
Price should be a float rather than a string

eport01 replied:
Also updated, thanks!

Yuji3000 commented on app/models/menu_item.rb:
```diff
@@ -0,0 +1,3 @@
+class MenuItem < ApplicationRecord
```
validations for our models?

eport01 replied:
Added validations and validation tests for our models!

eport01 committed:
update model validations

eport01 committed:
add menuitem validations

eport01 committed:
update seed data to reflect schema

eport01 commented:
Updated validations and tests for models, database to reflect tags instead of allergens, changed price to a float, and reseeded the database to reflect these changes. I also updated the server to localhost:3001 to reflect the json contract.

Yuji3000 reviewed (approved):
Thanks for making the changes!
